tasks/pkg-macos/macos_agent_version.yml

```
@@ -1,4 +1,6 @@
 ---
 - name: Set agent download filename to a specific version
   ansible.builtin.set_fact:
-    agent_dd_download_url: "{{ datadog_macos_versioned_url }}-{{ agent_datadog_agent_macos_version }}-1.dmg"
+    agent_dd_download_url: "{{ datadog_macos_versioned_url }}-{{ agent_datadog_agent_macos_version }}-1{{ maybe_arch }}.dmg"
+  vars:
+    maybe_arch: "{{ '.' + datadog_agent_macos_architecture if agent_datadog_agent_macos_version is version('7.70', '>=') else '' }}"
```

```suggestion
    agent_dd_download_url: "{{ datadog_macos_versioned_url }}-{{ agent_datadog_agent_macos_version }}-1{{ maybe_arch }}.dmg"
  vars:
    maybe_arch: "{{ '.' + (datadog_agent_macos_architecture | default(ansible_facts.architecture)) if agent_datadog_agent_macos_version is version('7.70', '>=') else '' }}"
```
To prevent an undefined variable error